api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceServiceImpl.java

```diff
@@ -189,4 +190,59 @@ public Workspace updateUserRoles(Workspace workspace, Set<WorkspaceUserRole> use
     }
     return this.saveWithLastModified(workspace);
   }
+  @Override
+  public WorkspaceAccessLevel getWorkspaceAccessLevel(String workspaceNamespace, String workspaceId) {
+    String userAccess;
+    try {
+      userAccess = fireCloudService.getWorkspace(
+          workspaceNamespace, workspaceId).getAccessLevel();
+    } catch (org.pmiops.workbench.firecloud.ApiException e) {
+      if (e.getCode() == 404) {
+        throw new NotFoundException(String.format("Workspace %s/%s not found",
```

Comment:
Just to clarify: this means if you don't have read access to a workspace and try to do something to it, you'll get a 404, and if you don't have write access and try to write to it, you'll get a 403 (below.)

I think that's fine and consistent with FC... just checking...

Reply:
Thats correct.